Objects/tupleobject.c

```diff
@@ -419,14 +431,32 @@ tupleitem(PyTupleObject *a, Py_ssize_t i)
     return a->ob_item[i];
 }
 
+static PyObject *
+tuple_from_array(PyObject *const *src, Py_ssize_t n, Py_ssize_t step)
```

Comment:
I dislike step. I prefer to leave tuplesubscript() unchanged and avoid step complication here.

Reply:
Any reason why this complexity better fits for `tuplesubscript()` rather than for `tuple_from_array()`?

Reply:
I prefer to keep tuple_from_array() simple. The compiler should be able to emit more efficient code without step.

The whole purpose of the change is performance, no?

Reply:
Well, others may have a different opinion, but I still don't think that adding a step argument is worth it for a single call site: tuplesubscript(). I would prefer a very straightforward tuple_from_array() without step argument: well, it becomes _PyTuple_FromArray() in this case. No need for a subfunction.